### PR TITLE
add support immutable class.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️

### CODE REVIEW

File: src/Definitions/ArrayBuilder.php [hunk 1]
```diff
@@ -151,7 +151,10 @@ private function configure(ContainerInterface $container, $object, iterable $con
         foreach ($config as $action => $arguments) {
             if (substr($action, -2) === '()') {
                 // method call
-                \call_user_func_array([$object, substr($action, 0, -2)], $arguments);
+                $setter = \call_user_func_array([$object, substr($action, 0, -2)], $arguments);
+                if (\is_object($setter)) {
```
Comment (yiiliveext) [inline src/Definitions/ArrayBuilder.php:155]: `if (\is_object($setter) && $setter instanceof $object) `

Comment (samdark) [reply src/Definitions/ArrayBuilder.php:155]: Yes. That is needed.

Comment (yiiliveext) [reply src/Definitions/ArrayBuilder.php:155]: Or just (supported by php7.4) `if ($setter instanceof $object)`

Comment (terabytesoftw) [reply src/Definitions/ArrayBuilder.php]: Done.

**Comment by samdark** [comment]
Thanks!

**Pull request merged**